Add documentation for subroutes and k8s service visibility tagging (#1827)
knative/docs: docs/serving/using-subroutes.md

---
title: "Creating and using Subroutes"
linkTitle: "Configuring cluster-local services"
weight: 20
type: "docs"
---

> Subroutes are most effective when used with multiple revisions. When defining a knative service/route, the traffic section of the spec can split between the different revisions. For example:

Reviewer: Global: capitalize Knative and Kubernetes.

```yaml
traffic:
- percent: 0
  revisionName: foo
- percent: 40
  revisionName: bar
- percent: 60
  revisionName: baz
```

This allows anyone targeting the main route to have a 0% chance of hitting revision `foo`, 40% chance of hitting revision `bar` and 60% chance of hitting revision `baz`.

> 

Reviewer: ```suggestion

## Using tags to create target URLs

```

Author: This has been added, Github didn't detect it.

> As part of the spec there's a key called `tag`. When a `tag` can be applied to the routes, this will generate an address for the specific traffic target.

Reviewer: ```suggestion
The spec defines a key called `tag`. When a `tag` is applied to a route, an address for the specific traffic target is created.
```

> As part of the spec there's a key called `tag`. When a `tag` can be applied to the routes, this will generate an address for the specific traffic target.

Reviewer: Is "key" the correct term? Would "attribute" be more accurate? (Not sure, but I don't recall seeing the term key used like this before, so I wanted to ask.)

Author: "attribute" makes more sense in terms of describing the spec. I'll update it. Thanks!

```yaml
traffic:
- percent: 0
  revisionName: foo
  tag: staging
- percent: 40
  revisionName: bar
- percent: 60
  revisionName: baz
```

> In the above example, we can access the staging target by accessing `staging-<route name>.<namespace>.<domain>`. Whereas the targets for `bar` and `baz` can only be accessible via the main route `<route name>.<namespace>.<domain>`.

Reviewer: ```suggestion
In the above example, you can access the staging target by accessing `staging-<route name>.<namespace>.<domain>`. The targets for `bar` and `baz` can only be accessed using the main route, `<route name>.<namespace>.<domain>`.
```
